### PR TITLE
Improve and unify page headers

### PR DESCRIPTION
**Pull request by dennisreimann**
As discussed in #2362, see the elaborated version in [this comment](https://github.com/btcpayserver/btcpayserver/discussions/2362#discussioncomment-525527).

### CODE REVIEW

**Comment by pavlenex** [comment]
tACK.

Just giving a bit more context on what this actually relates to:

## Before PR
Screenshot 2021-03-31 at 18 53 05
Screenshot 2021-03-31 at 18 51 07

## With PR applied
Screenshot 2021-03-31 at 18 52 34
Screenshot 2021-03-31 at 18 51 13

**Comment by dstrukt** [comment]
One thing I've been thinking about since our meeting is the notion of the sticky left sidebar, and the sticky heading element.

StickLeftSidebarSmall

One issue you might recognize immediately with the title, is if there is potentially a long title that extends beyond the left side navigation, it would overlap with the right nav, making the content difficult to read upon scroll.

Solutions:

1. We could potentially tighten up the margins, and add a background (same color) that all other content scrolls under, which would enable us to add always visible CTAs or buttons to the far top right section (as seen in the .gif) in the future.

2. Alternatively we could simply restrict the number of title characters to keep the title from becoming too long, but I don't know that I love this solution.

I'm still exploring better ways to display this information, but I like where this is headed.

Screen Shot 2021-03-31 at 1 39 48 PM

**Comment by dstrukt** [comment]
@dennisreimann and I briefly chatted about this PR, and we're going to wait to discuss any further changes, beyond what's in the PR, at the next design meeting, and open a new PR to address any further decisions.

File: BTCPayServer/Views/Server/SSHService.cshtml [hunk 1]
```diff
@@ -1,26 +1,22 @@
 ﻿@model BTCPayServer.Models.ServerViewModels.SSHServiceViewModel
 @{
-    ViewData.SetActivePageAndTitle(ServerNavPages.Services);
+    ViewData.SetActivePageAndTitle(ServerNavPages.Services, "SSH settings");
 }
 
+<h2 class="mb-4">@ViewData["PageTitle"]</h2>
 
-<h4>SSH settings</h4>
-<partial name="_StatusMessage" />
```
[the remark this answers is not in the document]
Comment (Kukks) [reply BTCPayServer/Views/Server/SSHService.cshtml:8]: nvm ^

**Review by Kukks — COMMENTED** [review]
I went through all the files. 

We cannot have PRs like this again. It is almost certain I skimmed through a file or 2 and lost my attention. 

It's also a major headache if this is merged before literally any other PR as it will conflict with everything.

File: BTCPayServer/Views/Server/SSHService.cshtml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dennisreimann) [reply BTCPayServer/Views/Server/SSHService.cshtml:8]: It's been moved to the NavLayout for consistency.

File: BTCPayServer/wwwroot/main/site.css [hunk 2]
```diff
@@ -110,12 +124,17 @@ a.nav-link:hover {
     margin-left: -1rem;
 }
 
-.nav-pills .nav-link.active,
```
[the remark this answers is not in the document]
Comment (dennisreimann) [reply BTCPayServer/wwwroot/main/site.css:113]: The `nav-pills` class isn't used anymore, that's why I removed it from the styles :)

**Comment by dennisreimann** [comment]
> We cannot have PRs like this again. It is almost certain I skimmed through a file or 2 and lost my attention.

Yes, I agree this is unfortunate as it affects every page view. We could split such updates into multiple chunks the next time, so that it is easier to review and also helps with the next thing you described …

> It's also a major headache if this is merged before literally any other PR as it will conflict with everything.

Yes, that's why I outlined the process [here](https://github.com/btcpayserver/btcpayserver/discussions/2362#discussioncomment-542353). I will provide the necessary updates/test fixes for this later today so that we can go ahead and merge it hopefully soon. Right now we don't have other UI related PRs open so that it would be a good timing to merge it this week to prevent what you described.

**Comment by dennisreimann** [comment]
Tests working and good to go from my side. As @dstrukt mentioned we'll do other potential additions in a separate PR.

**Comment by dstrukt** [comment]
We can adjust our font sizes, line-heights, etc.. in the future after we address the colors as well. They need a little bit of updating.

File: BTCPayServer/Views/EthereumLikeStore/GetStoreEthereumLikePaymentMethods.cshtml [hunk 3]
```diff
@@ -5,22 +5,13 @@
...
     Layout = "../Shared/_NavLayout.cshtml";
-
-    ViewData.SetActivePageAndTitle(StoreNavPages.ActivePage, "Ethereum Settings");
-
+    ViewData.SetActivePageAndTitle(StoreNavPages.ActivePage, "Ethereum Settings", Context.GetStoreData().StoreName);
     ViewData["NavPartialName"] = "../Stores/_Nav";
 }
 
-
-<partial name="_StatusMessage"/>
```
Comment (NicolasDorier) [inline BTCPayServer/Views/EthereumLikeStore/GetStoreEthereumLikePaymentMethods.cshtml:15]: Removed the statusMessage completely?

Comment (dennisreimann) [reply BTCPayServer/Views/EthereumLikeStore/GetStoreEthereumLikePaymentMethods.cshtml:15]: It has been moved to the outer NavLayout for consistency.

**Comment by NicolasDorier** [comment]
In the future, keep wide but dumb changes of the same type in a single PR so that's it's easy to review.

**Comment by NicolasDorier** [comment]
Ping @pavlenex @britttttk you want to give it a quick test?

File: BTCPayServer/Views/Wallets/WalletSendVault.cshtml [hunk 4]
```diff
@@ -1,19 +1,11 @@
 @model WalletSendVaultModel
 @{
     Layout = "../Shared/_NavLayout.cshtml";
-    ViewData["Title"] = "Manage wallet";
-    ViewData.SetActivePageAndTitle(WalletsNavPages.Send);
+    ViewData.SetActivePageAndTitle(WalletsNavPages.Send, "Sign the transaction with BTCPayServer Vault", Context.GetStoreData().StoreName);
```
Comment (britttttk) [inline BTCPayServer/Views/Wallets/WalletSendVault.cshtml:4]:
```suggestion
    ViewData.SetActivePageAndTitle(WalletsNavPages.Send, "Sign the transaction with BTCPay Server Vault", Context.GetStoreData().StoreName);
```

File: BTCPayServer/Views/Wallets/WalletTransactions.cshtml [hunk 5]
```diff
@@ -58,40 +57,35 @@
...
-                @foreach (var label in Model.Labels)
-                {
-                    <a asp-route-labelFilter="@label.Text" class="badge mr-2 my-1 position-relative text-white d-block"
-                       style="background-color: @label.Color;"><span class="text-white">@label.Text</span></a>
-                }
+<h4 class="mb-3">@ViewData["PageTitle"]</h4>
+<p>
+    If BTCPay Server shows you an invalid balance, <a asp-action="WalletRescan" asp-route-walletId="@Context.GetRouteValue("walletId")">rescan your wallet</a>.
+    <br/>
+    If some transactions appear in BTCPay Server, but are missing in another wallet, <a href="https://docs.btcpayserver.org/FAQ/FAQ-Wallet/#missing-payments-in-my-software-or-hardware-wallet">follow those instructions</a>.
```
Comment (britttttk) [inline BTCPayServer/Views/Wallets/WalletTransactions.cshtml:65]:
```suggestion
    If some transactions appear in BTCPay Server, but are missing in another wallet, <a href="https://docs.btcpayserver.org/FAQ/FAQ-Wallet/#missing-payments-in-my-software-or-hardware-wallet">follow these instructions</a>.
```

**Review by britttttk — COMMENTED** [review]
Reviewed code, I will test it tomorrow.

**Comment by dennisreimann** [comment]
Updated with master, hit it @NicolasDorier

**Comment by NicolasDorier** [comment]
I merge it for now, since I am worried about conflict with other PRs. The changes are safe in any case. But need testers check.

**Comment by Kukks** [comment]
relevant: https://tumblebit.slack.com/archives/C35LCNW93/p1617996517036400

**Comment by dstrukt** [comment]
Don't have an account with tumblebit .. any selective screenshots/copy worth sharing?

**Comment by dennisreimann** [comment]
@dstrukt it was a joke referring to large commits and merge conflicts.

**Comment by dstrukt** [comment]
Ooo.

😂

**Pull request merged**